src/components/common/PrivateRoute.tsx

```diff
@@ -0,0 +1,40 @@
+import { Navigate, useLocation } from 'react-router-dom';
+import type { ReactNode } from 'react';
+import { useAuth } from '@/hooks';
+
+interface PrivateRouteProps {
+  children: ReactNode;
+}
+
+const PrivateRoute = ({ children }: PrivateRouteProps) => {
+  const { isAuthenticated, loading } = useAuth();
+  const location = useLocation();
+
+  // 로딩 중일 때는 로딩 표시 (짧은 시간이므로 간단하게)
+  if (loading) {
+    return (
+      <div
+        style={{
+          display: 'flex',
+          justifyContent: 'center',
+          alignItems: 'center',
+          height: '200px',
+          fontSize: '16px',
+          color: '#666',
+        }}
+      >
+        로딩 중...
+      </div>
+    );
+  }
+
+  // 로그인되지 않은 경우 로그인 페이지로 리다이렉트
+  if (!isAuthenticated) {
+    return <Navigate to="/login" state={{ from: location.pathname }} replace />;
```

Comment:
App.tsx에서 상수로 선언한 ROUTE_LOGIN를 재사용하면 이 Navigate 컴포넌트의 to에도 재사용할 수 있을 것 같아요~

```suggestion
    return <Navigate to={ROUTE_LOGIN} state={{ from: location.pathname }} replace />;
```

Reply:
App.tsx에서 선언한 상수를 export한 후 다른 파일에서 import하여 사용하도록 구현하는게 뭔가 부자연스러워서, constants/routes.ts파일로 경로만 따로 분리한 후 여기서 import하도록 수정했습니다.

경로를 처리할 때 더 좋은 방식이 있을까요?

Reply:
잘 분리하신 것 같아요!